background/loadScripts.js

```diff
@@ -43,36 +44,52 @@ export default function loadScripts (port) {
     }
 
     if (!codeCache) {
-      const vendor = fetch(`http://${UI}/vendor.js`)
-        .then(checkResponseOk)
-        .then(response => response.blob());
-      const embed = fetch(`http://${UI}/embed.html`)
+      codeCacheVersion = State.version;
+
+      codeCache = fetch(`http://${UI}/embed.json`)
         .then(checkResponseOk)
-        .then(response => response.text())
-        .then(page => ({
-          styles: /styles\/embed\.([a-z0-9]{10})\.css/.exec(page),
-          scripts: /embed\.([a-z0-9]{10})\.js/.exec(page)
-        }))
-        .then(res => {
-          return Promise.all([
-            fetch(`http://${UI}/${res.styles[0]}`).then(checkResponseOk),
-            fetch(`http://${UI}/${res.scripts[0]}`).then(checkResponseOk)
-          ]);
+        .then((response) => response.json())
+        .then((embed) => {
+          const { assets } = embed;
+
+          const filteredAssets = assets
+            .filter((asset) => !/html$/.test(asset))
+            .filter((asset) => !/^embed(.+)js$/.test(asset));
+
+          const mainScript = assets.find((asset) => /^embed(.+)js$/.test(asset));
+
+          const assetsPromises = filteredAssets
+            .map((asset) => {
+              return fetch(`http://${UI}/${asset}`)
+                .then((response) => response.blob())
+                .then((blob) => {
+                  if (/\.js$/.test(asset)) {
+                    return URL.createObjectURL(blob, { type: 'application/javascript' });
+                  }
+
+                  return URL.createObjectURL(blob);
+                })
+                .then((url) => ({ path: asset, url }));
+            });
+
+          const scriptPromise = fetch(`http://${UI}/${mainScript}`)
+            .then((response) => response.text());
+
+          return Promise.all([ scriptPromise, Promise.all(assetsPromises) ]);
         })
-        .then(responses => Promise.all(responses.map(response => response.blob())));
+        .then(([ script, assets ]) => {
+          assets.forEach((asset) => {
```

Comment:
Ah, this. Well right now there aren't any really useful assets (no images, etc.), but the idea is that the script might load some images/fonts/etc... If we don't replace the URLs of theses assets, it will try to load, let's say, `/image.png`. This doesn't exist in the Chrome Extension background environment, so it will fail. The best approach IMO would have been to be able to catch these requests and answer with the right URL or already fetched asset. However, Chrome Extensions can't have Service Worker, as you explained, and it cannot catch and modify requests made from within the extension.

So this is the only solution left that I found :/ Not the best though, I agree.